tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/EpicsControllerTests.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Common;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NHSD.BuyingCatalogue.Solutions.API.Controllers;
+using NHSD.BuyingCatalogue.Solutions.API.ViewModels.Epics;
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.UpdateClaimedEpics;
+using NHSD.BuyingCatalogue.Solutions.Application.Commands.Validation;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Epics;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Solutions.API.UnitTests
+{
+    public sealed class EpicsControllerTests : ControllerBase
+    {
+        private Mock<IMediator> _mockMediator;
+        private EpicsController _controller;
+        private const string SolutionId = "Sln1";
+
+        [SetUp]
+        public void Setup()
+        {
+            _mockMediator = new Mock<IMediator>();
+            _controller = new EpicsController(_mockMediator.Object);
+        }
+
+        [Test]
+        public async Task ShouldUpdateValidationValidAsync()
+        {
+            HashSet<ClaimedEpicViewModel> claimedEpics = new HashSet<ClaimedEpicViewModel>()
+            {
+                new ClaimedEpicViewModel()
+                {
+                    EpicId = "Epic1",
+                    StatusName = "Passed"
+                }
+            };
+
+            var updateEpicsViewModel = new UpdateEpicsViewModel {ClaimedEpics = claimedEpics};
+
+            var epicsViewModel = new HashSet<IClaimedEpic>(updateEpicsViewModel.ClaimedEpics);
+
+            var viewModel = new UpdateEpicsViewModel {ClaimedEpics = claimedEpics};
+            var validationModel = new Mock<ISimpleResult>();
+            validationModel.Setup(s => s.IsValid).Returns(true);
+
+            _mockMediator
+                .Setup(m => m.Send(It.Is<UpdateClaimedEpicsCommand>(e => e.SolutionId == SolutionId),
+                    It.IsAny<CancellationToken>())).ReturnsAsync(validationModel.Object);
+
+            var result = (await _controller.UpdateAsync(SolutionId, viewModel).ConfigureAwait(false)) as NoContentResult;
+            result.StatusCode.Should().Be((int)HttpStatusCode.NoContent);
+
+            _mockMediator.Verify(
+                m => m.Send(It.Is<UpdateClaimedEpicsCommand>(e => e.SolutionId == SolutionId),
```

Comment:
Need to come up with a way to ensure the list of epics is what we expect in the verify. One example would be to use an equality comparer or use a call back.

Reply:
Task created in the backlog #5069